atomic_reactor/plugins/post_fetch_docker_archive.py

```diff
@@ -0,0 +1,51 @@
+"""
+Copyright (c) 2015-2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import os
+
+from atomic_reactor.constants import (EXPORTED_SQUASHED_IMAGE_NAME, IMAGE_TYPE_DOCKER_ARCHIVE)
+from atomic_reactor.dirs import BuildDir
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.util import get_exported_image_metadata, is_scratch_build
+
+
+class FetchDockerArchivePlugin(PostBuildPlugin):
+    key = 'fetch_docker_archive'
+    is_allowed_to_fail = False
+
+    def __init__(self, workflow):
+        """
+        :param workflow: DockerBuildWorkflow instance
+        """
+        super(FetchDockerArchivePlugin, self).__init__(workflow)
+        self.source_build = bool(self.workflow.data.build_result.source_docker_archive)
+
+    def download_image(self, build_dir: BuildDir):
+        image = self.workflow.data.tag_conf.get_unique_images_with_platform(build_dir.platform)[0]
```

Comment:
here I am assuming the list is of length one and contains the images we're want to download. Please correct me if wrong.

Reply:
That sounds correct :+1: 

Even if there somehow were multiple unique tags, we would expect all of them to point to the same image, so taking any one of them should be fine.